src/backend/joanie/payment/backends/base.py

```diff
@@ -53,15 +55,46 @@ def _do_on_payment_success(cls, order, payment):
 
         order.set_installment_paid(payment["installment_id"])
 
-        # send mail
-        cls._send_mail_payment_success(order)
+        if order.get_position_last_paid_installment() == 1 or (
+            order.payment_schedule_fully_paid() and len(order.payment_schedule) == 1
+        ):
+            cls._send_mail_payment_success(order=order)
```

Comment:
This mail should not be sent when the order is fully paid but when this one is validated so when the order goes to `pending` state.

Reply:
This will never be triggered. Indeed, here you will receive confirmation of an installment payment so the order always be transitionned from pending to `failed_payment` or `pending_payment` This email should be sent when the order transition from `to_save_payment_method` | `signing` to `pending`